### ISSUE TITLE
Serverless v4 pricing change

### ISSUE THREAD

**Issue by sreepathy-tl** [brefphp/bref#1749]
You must be aware of the recent [pricing change for Serverless CLI](https://www.serverless.com/pricing). Tightly coupled usage of serverless cli will make bref less acceptable to people who are affected by the pricing change. Please consider providing an alternative to the serverless cli integration like AWS SAM / Terraform based deployment.

Since there are not many mature alternatives to bref, please consider this request.

**Comment by rafaelbernard** [comment]
I have been writing about using Bref with CDK. [Constructs have reached v1](https://github.com/brefphp/constructs/releases). Although I am not using it in the Production workload yet (I will very soon), everything is mature and stable.

https://dev.to/rafaelbernard/a-bref-aws-php-history-part-1-2agn
https://dev.to/rafaelbernard/a-bref-aws-php-story-part-2-1dhe
https://dev.to/rafaelbernard/a-bref-aws-php-story-part-3-147f

**Comment by kevincerro** [comment]
I've +10 projects using bref
This means that now I will need to pay for serverless framework usage?

**Comment by kevincerro** [comment]
Ok I've readed in their blog that this doesn't affect to SLS Framework V3 or companies under 2M in revenue
https://www.serverless.com/blog/serverless-framework-v4-a-new-model

**Comment by mnapoli** [comment]
Yep, the threshold is $2M in annual revenue, so for some Bref users there will be no impact. I mean, unless the project actually changes stuff that impacts the Bref plugin 🤷

V4 is not out yet, so it's hard to have actual details on the pricing and what this new version is about. It was supposed to be out 3 weeks ago, no news. I'm watching it on GitHub but there are very little details nor commits. Until we have more info it's really hard to get an idea, I don't know what to expect at all.

**Comment by gavinlove** [comment]
The pricing has now been published, along with an FAQ. My understanding is that even if we have a lambda function that gets no traffic in the month because it is provisioned, we would need to buy credit for it. We use Bref a lot but we currently don't use the serverless.com Dashboard at all and the pricing is very much aimed at using their dashboard.

https://www.serverless.com/pricing

**Comment by austencollins** [comment]
Hi @gavinlove, the pricing works differently than you described. The CLI is free to developers and organizations with less than $2M in annual revenue. The Dashboard is a separate paid set of features, which is not required by the CLI.

For those above the CLI revenue threshold, the cost is 1 Credit (~$3) per deployed instance of a Service, as defined by a `serverless.yml` file, which can include _one_ or _many_ AWS Lambda functions. To be clear, we do not charge based on individual AWS Lambda functions. The use of "Service" in Serverless Framework refers to the entire project file, which might have caused some confusion.

**Comment by fredericgboutin-yapla** [comment]
For the posterity - https://github.com/oss-serverless/osls

Kudos to @mnapoli for that move 🎉 🙏
